### PR TITLE
[#14] [Backend] As a user, I can view the list of keywords and their results uploaded by CSV files

### LINKED ISSUE #14: [Backend] As a user, I can view the list of keywords and their results uploaded by CSV files
## Why

User needs to see all the records from their previously uploaded CSV

## Who Benefits?

The user who needed to use view and filter the results from uploaded CSV

### PR DESCRIPTION
**Pull request by nvminhtue**
close https://github.com/nvminhtue/elixir_internal_certificate/issues/14

## What happened

Showing list and detail of scraping keywords
 
## Insight

- Refactor view of list and details of scraping keywords
- Paginate returns list results
- Show detail scraped keyword page from `success` status `user_search` record
 
## Proof Of Work
List of `user_search` keywords
image

Detail of a specific `search_result`
image

Showing the result of html_response
image

### CODE REVIEW

File: lib/elixir_internal_certificate_web/router.ex [hunk 1]
```diff
@@ -82,9 +82,9 @@ defmodule ElixirInternalCertificateWeb.Router do
   scope "/", ElixirInternalCertificateWeb do
     pipe_through [:browser, :require_authenticated_user]
 
-    get "/", UserSearchController, :index
+    resources "/keywords", UserSearchController, only: [:index, :show]
 
-    get "/keywords/:id", UserSearchController, :show
+    get "/", PageController, :index
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate_web/router.ex:87]: not sure the purpose of that controller so far :D

File: lib/elixir_internal_certificate_web/templates/user_search/_keyword_list.html.heex [hunk 2]
```diff
@@ -1,64 +1,29 @@
...
-      </td>
-    </tr>
+    <%= for {user_search, i} <- Enum.with_index(@data, 1) do %>
+      <tr role="button" class="align-middle" key={i}>
+        <td><%= user_search.keyword %></td>
+        <td class={get_status(user_search.status)}><%= user_search.status %></td>
+        <td>
+          <%= if user_search.status == :success do %>
+            <%=
+              link "Click for detail",
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate_web/templates/user_search/_keyword_list.html.heex]: Can we just use `Details` 🤔

File: lib/elixir_internal_certificate/scraper/scrapers.ex [hunk 3]
```diff
@@ -35,6 +32,25 @@ defmodule ElixirInternalCertificate.Scraper.Scrapers do
     |> Repo.insert(returning: true)
   end
 
+  def get_user_search(user_id, id) when is_integer(id) or is_binary(id) do
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate/scraper/scrapers.ex]:
```suggestion
  def get_user_search_by_user_id_and_id(user_id, id) when is_integer(id) or is_binary(id) do
```

File: test/elixir_internal_certificate/scraper/scrapers_test.exs [hunk 4]
```diff
@@ -230,4 +212,237 @@ defmodule ElixirInternalCertificate.Scraper.ScrapersTest do
...
+    test "given an INVALID user_search id, returns error" do
+      insert(:user_search)
+
+      assert_raise FunctionClauseError, fn ->
+        Scrapers.get_user_search(:invalid)
+      end
+    end
+  end
+
+  describe "get_user_searches/3" do
```
Comment (andyduong1920) [inline test/elixir_internal_certificate/scraper/scrapers_test.exs]:
```suggestion
  describe "get_user_searches/2" do
```

File: test/elixir_internal_certificate/scraper/scrapers_test.exs [hunk 5]
```diff
@@ -230,4 +212,237 @@ defmodule ElixirInternalCertificate.Scraper.ScrapersTest do
              ]
     end
   end
+
+  describe "get_user_search/1" do
```
Comment (andyduong1920) [inline test/elixir_internal_certificate/scraper/scrapers_test.exs]: seem we are missing the test for `get_user_search/2`

File: lib/elixir_internal_certificate_web/router.ex (hunk 1 above)
Comment (nvminhtue) [reply lib/elixir_internal_certificate_web/router.ex:87]: I want to redirect all request from "{host_url}" to "{host_url}/keywords" @andyduong1920

File: test/elixir_internal_certificate/scraper/scrapers_test.exs (hunk 5 above)
Comment (nvminhtue) [reply test/elixir_internal_certificate/scraper/scrapers_test.exs]: Good catch, updated in [ba8fc3c](https://github.com/nvminhtue/elixir_internal_certificate/pull/37/commits/ba8fc3c46630332abcc5a4027398965f1cbc898e)

**Review by andyduong1920 — APPROVED** [review]

**Review by junan — APPROVED** [review]

**Pull request merged**